app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,39 @@
+class CategoriesController < ApplicationController
+  before_action :authenticate_user!, except: [:splash]
+
+  ICON_OPTIONS = ['🍔', '🛒', '🚗', '🐶', '📚', '💻', '🎁', '🏠', '🏥', '🎬', '👔', '🎓'].freeze
+
+  def index
+    @categories = current_user.categories.includes(:expenses)
+  end
+
+  def new
+    @category = Category.new
+    @options = ICON_OPTIONS
+  end
+
+  def create
+    @category = current_user.categories.build(categories_params)
+
+    if @category.save
+      redirect_to categories_path, notice: 'Expense category was successfully created.'
+    else
```

Comment:
- [ ] dear @kit0-0, please keep in mind we need to keep our inputs validated and make sure our app does not crash when an invalid input is submitted. This validation is done both from the HTML and the Rails side. currently, when I create a category by setting the name without selecting an icon, an error is thrown, and the app crashes. Please take a look into this. 

### The issue 

https://github.com/kit0-0/budget/assets/69077061/ec3bf1a1-4d29-4e5e-90a2-6c5e536ede27

- [ ] This issue persists when I try to create an expense without checking any of the checkboxes, Please consider addressing this too.